internal/restconfig/restconfig.go

```
@@ -271,6 +263,10 @@ func (rcp *Producer) RunOnSelectedContext(function PerContextFn, status reporter
 		return status.Error(err, "error building the cluster.Info for the default configuration")
 	}
 
+	if clusterInfo.Submariner != nil {
+		checkVersionMismatch(clusterInfo.Submariner)
```

You're ignoring the error completely...